OpenRA.Mods.Common/Traits/Transforms.cs

```diff
@@ -127,11 +128,21 @@ public void DeployTransform(bool queued)
 				Faction = faction
 			};
 
+			var checkAndTransform = new CallFunc(() =>
+			{
+				// We locked it at the start of this method, so we need to unlock it now.
+				if (building != null)
+					building.Unlock();
```

Comment:
This leaves the actor unlocked for a tick, which opens us up to a bunch of edge cases (e.g. selling / capturing the conyard immediately before it turns into an MCV).

You don't need to unlock the actor because it gets discarded when the transform succeeds.  I would set this up like:

```
queue(turn)
queue(lock)
queue(play unmake animation)
queue(check footprint, and transform if allowed)
queue(make animation)
queue(unlock)
```

If the transform succeeds then the activities after that won't run.  If the transform fails then it will restore itself to a valid state.